src/libmongoc/tests/test-mongoc-versioned-api.c

```diff
@@ -174,6 +174,37 @@ _test_mongoc_server_api_client_pool_once (void)
    mongoc_uri_destroy (uri);
 }
 
+static void
+_test_mongoc_client_uses_server_api (void)
+{
+   mongoc_client_t *client0; /* versioned */
+   mongoc_client_t *client1; /* not versioned */
+   mongoc_server_api_t *api;
+   bson_error_t error;
+
+   client0 = mongoc_client_new ("mongodb://localhost");
+   client1 = mongoc_client_new ("mongodb://localhost");
```

Comment:
This should use to `test_framework` APIs to create a client, so that the client behavior can be tweaked by the test runner.

Reply:
The next section of code in the test asserts that no server API has been set, and then the test goes on to set the API on one and only one client. Is there a test_framework_* function that will still let me do that?

Reply:
I do not think there is one. I suggest defining a `test_framework_client_pool_new_no_server_api`.